tests/components/media_player/test_directv.py

```diff
@@ -0,0 +1,445 @@
+        self._device = clientAddr
+        self._standby = True
+        self._play = False
+
+        _LOGGER.debug("INIT with host: %s, port: %s, device: %s",
+                      self._host, self._port, self._device)
+
+        self._locations = LOCATIONS
+
+        self.attributes = LIVE
+
+    def get_locations(self):
+        """Mock for get_locations method."""
+        _LOGGER.debug("get_locations called")
+        test_locations = {
+            'locations': self._locations,
+            'status': {
+                'code': 200,
+                'commandResult': 0,
+                'msg': 'OK.',
+                'query': '/info/getLocations'
+            }
+        }
+
+        return test_locations
+
+    def get_standby(self):
+        """Mock for get_standby method."""
+        _LOGGER.debug("STANDBY is: %s", self._standby)
+        return self._standby
+
+    def get_tuned(self):
+        """Mock for get_tuned method."""
+        _LOGGER.debug("get_tuned called")
+        if self._play:
+            self.attributes['offset'] = self.attributes['offset']+1
+
+        test_attributes = self.attributes
+        test_attributes['status'] = {
+            "code": 200,
+            "commandResult": 0,
+            "msg": "OK.",
+            "query": "/tv/getTuned"
+        }
+        return test_attributes
+
+    def key_press(self, keypress):
+        """Mock for key_press method."""
+        _LOGGER.debug("Key Press: %s", keypress)
+        if keypress == 'poweron':
+            self._standby = False
+            self._play = True
+        elif keypress == 'poweroff':
+            self._standby = True
+            self._play = False
+        elif keypress == 'play':
+            self._play = True
+        elif keypress == 'pause' or keypress == 'stop':
+            self._play = False
+
+    def tune_channel(self, source):
+        """Mock for tune_channel method."""
+        _LOGGER.debug("Change channel %s", source)
+        self.attributes['major'] = int(source)
+
+
+class TestSetupDirectvMediaPlayer(unittest.TestCase):
+    """Test the different possibilities for setting up DirecTV media player."""
+
+    def setUp(self):
+        """Set up things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+
+        # Mocking DIRECTV class in DirectPy with our own.
+        self.directpy_mock = MagicMock()
+        modules = {
+            'DirectPy': self.directpy_mock
+        }
+        self.module_patcher = patch.dict('sys.modules', modules)
+        self.module_patcher.start()
+        import DirectPy
+        DirectPy.DIRECTV = mockDIRECTVClass
+
+        self.addCleanup(self.tearDown)
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+        self.module_patcher.stop()
+
+    def test_setup_platform_config(self):
+        """Test setting up the platform from configuration."""
+        add_entities = Mock()
+        setup_platform(
```

Comment:
Don't set up the platform directly. Set up the component with `async_setup_component` and pass the config with the media_player component and the directtv platform.

Reply:
Done :-)